### DISCUSSION TITLE
Running monitor 8 sidecar and dotnet app in k8s

### DISCUSSION THREAD

**Discussion by vinvli** [dotnet/dotnet-monitor#5424] [Q&A] [answered]
Hi,

We tested the pizza app (https://github.com/MicrosoftDocs/mslearn-dotnet-kubernetes) together with the dotnet monitor 8 on our kubernetes cluster and it seems to work fine. Port forwarding to 52323 and using postman for http://localhost:52323/processes and the like. Very impressed with the functionality. Ideal for things like prod troubleshooting.

However when we try it with our own home made dotnet app we can't seem to connect. We get the error "The runtime has been configured to pause during startup and is awaiting a Diagnostics IPC resumestartup command from a diagnostic port" and "System.ArgumentException: Unable to discover a target process". We use .net 6 and  FROM mcr.microsoft.com/dotnet/aspnet:6.0-alpine3.16. Same settings as documented for monitor 8

Any idea what could be the reason it goes into this suspend mode?

**Comment by jander-msft** [comment, marked as answer]
> We tested the pizza app (https://github.com/MicrosoftDocs/mslearn-dotnet-kubernetes) together with the dotnet monitor 8 on our kubernetes cluster and it seems to work fine. Port forwarding to 52323 and using postman for http://localhost:52323/processes and the like. Very impressed with the functionality. Ideal for things like prod troubleshooting.

That's great to hear!

> However when we try it with our own home made dotnet app we can't seem to connect. We get the error "The runtime has been configured to pause during startup and is awaiting a Diagnostics IPC resumestartup command from a diagnostic port" and "System.ArgumentException: Unable to discover a target process". We use .net 6 and FROM mcr.microsoft.com/dotnet/aspnet:6.0-alpine3.16. Same settings as documented for monitor 8
> 
> Any idea what could be the reason it goes into this suspend mode?

Even when it is all working correctly, your application is briefly suspended at launch to allow .NET Monitor to perform diagnostics from startup, assuming you've set the `DOTNET_DiagnosticPorts` environment variable on the target application container. .NET Monitor will immediately resume it once it has detected the application and started any collection rules you may have configured.

Assuming you've taken the successful deployment configuration from when you were testing with the pizza app and just swapped out the image to use your own application image, it should work the same.

Would you be able to share your deployment where you see the app is still suspended and .NET Monitor cannot find the process?

**Comment by sbonds** [comment]
To pile on to this rare Google double-hit item, I ran into this same error message, but only in a "real" environment. Of course, the small test environment I was using worked great.

The issue was in the pod `SecurityContext`. The quick fix is to delete the whole section so your containers run as root, have access to the host process space, and don't have any read only filesystems. It's probably possible to make this work using less draconian removals, but I'll take working over perfect tonight.
